apps/tangle-dapp/containers/TransferTxContainer/TransferTxContainer.tsx

```diff
@@ -122,28 +143,52 @@ const TransferTxContainer: FC<TransferTxContainerProps> = ({
               {txError.message}
             </Typography>
           )}
+
+          {!isValidReceiverAddress && receiverAddress !== '' && (
+            <Typography variant="body1" fw="normal" className="!text-red-50">
+              Invalid receiver address
+            </Typography>
+          )}
         </div>
 
-        <ModalFooter className="flex flex-col gap-1 px-8 py-6">
-          <Button
-            isFullWidth
-            isDisabled={!canInitiateTx || executeTransferTx === null}
-            isLoading={!isReady}
-            loadingText={getTxStatusText(status)}
-            onClick={executeTransferTx !== null ? executeTransferTx : undefined}
-          >
-            Send
-          </Button>
-
-          <Link href={TANGLE_DOCS_URL} target="_blank">
-            <Button isFullWidth variant="secondary">
-              Learn More
+        <ModalFooter className="flex items-center gap-2 px-8 py-6 space-y-0">
+          <div className="flex-1">
+            <Button
+              isFullWidth
+              isDisabled={
+                !canInitiateTx ||
+                executeTransferTx === null ||
+                !isValidReceiverAddress
+              }
+              isLoading={!isReady}
+              loadingText={getTxStatusText(status)}
+              onClick={
+                executeTransferTx !== null ? executeTransferTx : undefined
+              }
+            >
+              Send
             </Button>
-          </Link>
+          </div>
+
+          <div className="flex-1">
+            <Link href={TANGLE_DOCS_URL} target="_blank" className="w-full">
+              <Button isFullWidth variant="secondary">
+                Learn More
+              </Button>
+            </Link>
+          </div>
         </ModalFooter>
       </ModalContent>
     </Modal>
   );
 };
 
 export default TransferTxContainer;
+
+/** @internal */
+function getTypedChainIdFromAddr(addr?: string | null): number {
+  if (!addr) return NaN;
+  if (isEthereumAddress(addr)) return PresetTypedChainId.TangleTestnetEVM;
+  if (isAddress(addr)) return PresetTypedChainId.TangleTestnetNative;
+  return NaN;
```

Comment:
I think at this point an error should be thrown: if it's not a Substrate nor EVM address, then something's wrong. Returning NaN propagates that erroneous program state until something elsewhere fails. It's better to explicitly fail as early as possible